connection.js

```diff
@@ -1495,7 +1495,7 @@ class Connection {
             header = header.concat(this.transaction.notes.authentication_results);
         }
         if (header.length === 1) return '';  // no results
-        return header.join('; ');
+        return header.join(";\n\t");
```

Comment:
If you're going to fold this line, don't you need to use a FWS?

Reply:
Not sure what you're implying.

     authres-header = "Authentication-Results:" [CFWS] authserv-id
                  [ CFWS authres-version ]
                  ( no-result / 1*resinfo ) [CFWS] CRLF
     resinfo = [CFWS] ";" methodspec [ CFWS reasonspec ]
                   *( CFWS propspec )

WSP = `space (SP, ASCII value 32) and horizontal tab (HTAB,   ASCII value 9) characters`

I guess I inserted a "CRLF" before WSP, which is just "\t" in this case. And "\n" is gonna be translated later into \r\n by some header magic I don't care of. Anyway, I've done [what you've done](https://github.com/haraka/Haraka/blob/master/connection.js#L1463) right above this code: 
```js
received_header += `;\n\t${utils.date_to_str(new Date())}`
```

Reply:
I asked, "don't you need to use a FWS?"

The implication is that you didn't use FWS. You've added a `\n\t`, which does not include a FWS, because per the definition, a FWS uses a CRLF.

> And "\n" is gonna be translated later

Is there a reason to depend on that instead of using a CRLF here?

Reply:
CFWS           =   (1*([FWS] comment) [FWS]) / FWS
    FWS            =   ([*WSP CRLF] 1*WSP)
    WSP            =   " " / "\t"

For `header = ['x1', 'x2']`, the final result is going to be: `Authentication-Results: x1;\r\n\tx2`. Tell me what's not clear. Should I change the "\n" to "\r\n"? If so, why the code excerpt I've provided doesn't contain it?

Reply:
> Should I change the "\n" to "\r\n"? 

I would. Because it's more clear. Someone doesn't need to know about the later conversion in order to recognize that is a correct FWS.

> If so, why the code excerpt I've referenced earlier

That code was added many years ago by someone who was intimately familiar with that later conversion. I'm a fan of moving the cookie jar down to lower shelves.

Reply:
you asked, we delivered